src/storage.rs

```diff
@@ -1,15 +1,61 @@
 use std::fmt::Debug;
 
+use ahash::AHashMap;
 use alloy_primitives::{Address, Bytes, B256, U256};
 use revm::{
+    db::PlainAccount,
     primitives::{AccountInfo, Bytecode},
     DatabaseRef,
 };
 
+/// An account stored in memory.
+#[derive(Debug, Default, Clone, PartialEq)]
+pub struct EvmAccount {
+    /// The account's basic information.
+    pub basic: AccountBasic,
+    /// The account's storage.
+    pub storage: AHashMap<U256, U256>,
+}
+
+impl From<PlainAccount> for EvmAccount {
+    fn from(account: PlainAccount) -> Self {
+        EvmAccount {
+            basic: account.info.into(),
+            storage: account.storage.into_iter().collect(),
+        }
+    }
+}
+
+impl EvmAccount {
+    /// Checks if the account is empty.
+    /// An account is considered empty if its code is empty, balance is zero, and nonce is zero.
+    pub fn is_empty(&self) -> bool {
+        self.basic.code.is_empty() && self.basic.balance == U256::ZERO && self.basic.nonce == 0
+    }
+
+    /// Converts a `revm::primitives::Account` into an `Option<InMemoryAccount>`.
+    /// Returns `Some(InMemoryAccount)` if the account is not self-destructed, otherwise returns `None`.
+    pub fn from_revm_account(account: revm::primitives::Account) -> Option<Self> {
```

Comment:
I'd inline this into `transform_output` as it's not used anywhere else and the context is non-trivial. After inlining we can remove the ad-hoc `assert`, turn the actual conversion into a `From` instance, and de-risk misusage (using this function to convert between account primitives outside of a state transition context).